### DISCUSSION TITLE
How to embed figures from .qmd files?

### DISCUSSION THREAD

**Discussion by mpf** [quarto-dev/quarto-cli#8468] [Q&A] [answered]
### Description

[This is a follow-up to [this question](https://github.com/quarto-dev/quarto-cli/discussions/8400#discussion-6123491) on embeds, which went unresolved. Link to MWE at end of post.]

According to the documentation for [embedding from Quarto documents](https://quarto.org/docs/authoring/notebook-embed.html#quarto-documents), it should be possible to embed figures generated from within other `qmd` files, eg,

`index.qmd`:
````qmd
---
title: "Quarto MWE"
format: html
---

```{julia}
#| echo: true  # <-- preview throws error if 'false'
#| label: fig-index-plot
#| output: true
#| fig-cap: "A plot of the sine function"
using Plots
plot(sin, -2π, 2π)
```
````
`about.qmd`:
````qmd
---
title: "About"
format: html
---

{{< embed index.qmd#fig-index-plot >}}
````

But when I do this, the embed reproduces the code but not the figure, eg,

<img width="487" alt="Screenshot 2024-01-27 at 11 44 51 AM" src="https://github.com/quarto-dev/quarto-cli/assets/109942/f1755cda-cf99-4034-b58a-799117f59f6a">

Moreover, setting `echo: false` in the source cell causes preview to throw an error:
```
Rendering qmd embeds
[1/1] index.qmd

Executing 'index.ipynb'
  Cell 1/1: 'fig-index-plot'...Done

ERROR: The cell fig-index-plot does not exist in notebook

Stack trace:
    at file:///Applications/quarto/bin/quarto.js:69687:23
    at Array.map (<anonymous>)
    at notebookMarkdown (file:///Applications/quarto/bin/quarto.js:69684:40)
    at eventLoopTick (ext:core/01_core.js:183:11)
    at async replaceNotebookPlaceholders (file:///Applications/quarto/bin/quarto.js:69628:32)
    at async renderPandoc (file:///Applications/quarto/bin/quarto.js:69987:28)
    at async Object.onRender (file:///Applications/quarto/bin/quarto.js:77261:36)
    at async renderFileInternal (file:///Applications/quarto/bin/quarto.js:77238:17)
    at async renderFiles (file:///Applications/quarto/bin/quarto.js:77062:17)
    at async renderProject (file:///Applications/quarto/bin/quarto.js:77389:25)
```

1. Is it possible to include figures (but not code) from other qmd files?
2. Is it possible to hide the code in the source qmd file?

Here's a repo with a minimal working example, and also quarto version I'm using:
- [github repo](https://github.com/mpf/quarto-mwe.git)
- Quarto version 1.4.549

**Comment by mcanouil** [comment]
@mpf Why exactly are you opening a new discussion, if the issue is the same and it is a follow up?
Please do not open new discussions for no reason as currently it might seems like you were impatient because you did not get a solution within your expectations (3 work days).
Note that you got an answer, but not a solution because your initial report did not contain the necessary information to help you in the first place.

**Reply by mpf** [reply]
@mcanouil I didn't intent to push boundaries. I posted this related question because I wasn't able to earlier provide a MWE for my earlier question regarding the error thrown by `quarto preview` about not finding the "embed" command.  For this (presumably related) second case, I was able to create an MWE.

So, I was only trying to decouple the two issues. Sorry for the confusion.

For the question posted here, I can find a workaround until the next version by placing computations in a `ipynb` file. But for the earlier post with the `embed` error, I've yet no workaround and can't at all embed in revealjs format files.

**Reply by mcanouil** [reply]
@mpf FYI, it's fine. It's just that this means to spend more time since the information is in multiple places.
For instance, if a user find the other thread, he/she will have to switch to another thread, then to the issue, to get the end of the story, so not very convenient.

**Comment by mcanouil** [comment, marked as answer]
It seems like a bug, but you should not use `keep-ipynb` to embed the notebook.
This file is an intermediary file. `keep-` options are for debugging.
If you want to use a Jupyter Notebook out of a Quarto document, set ipynb as format.

Regarding your error message when using the Jupyter Notebook, since the cells were not executed, there is nothing in it you can use to embed.

> ERROR: The cell fig-index-plot does not exist in notebook

I opened an issue:
- https://github.com/quarto-dev/quarto-cli/issues/8472
